utils/dependencies.go

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright (c) 2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package utils
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/eclipse/che-plugin-broker/model"
+)
+
+// GetExtensionCollisions checks a list of plugin metas for extensions shared by
+// more than one plugin. Return value is a list of
+func GetExtensionCollisions(metas []model.PluginMeta) map[string][]string {
+	extensions := map[string][]string{}
+	for _, meta := range metas {
+		for _, ext := range meta.Spec.Extensions {
+			extensions[ext] = append(extensions[ext], meta.ID)
+		}
+	}
+	collisions := make(map[string][]string)
+	for ext, ids := range extensions {
+		if len(ids) > 1 {
+			collisions[ext] = ids
+		}
+	}
+	return collisions
+}
+
+// ConvertCollisionsToLog converts the ouput of GetExtensionCollisions to a human-readable
+// string. Output is a slice of strings, to be joined by newlines
+func ConvertCollisionsToLog(collisions map[string][]string) []string {
+	var output []string
+	for ext, plugins := range collisions {
+		output = append(output,
+			fmt.Sprintf("    Plugins"),
```

Comment:
Does it make sense to cover it with tests to make sure that there are no typos in the output?
Sometimes I figure out them in messages during test writing.

Also, I wonder what is the better format:
```
  Plugins 
    {ms-kubernetes-tools/vscode-kubernetes-tools/0.1.17, redhat/vscode-openshift-connector/0.1.2} 
  conflict on extension 
    https://github.com/redhat-developer/vscode-yaml/releases/download/0.4.0/redhat.vscode-yaml-0.4.0.vsix
```
vs
```
  Plugins 
    - ms-kubernetes-tools/vscode-kubernetes-tools/0.1.17
    - redhat/vscode-openshift-connector/0.1.2
  conflict on extension 
    https://github.com/redhat-developer/vscode-yaml/releases/download/0.4.0/redhat.vscode-yaml-0.4.0.vsix
```

Reply:
I'm generally against testing log messages for exact structure, since it doesn't really do much once they're written; any change to the log message means blindly mirroring the message to the test. Either way, I've added a test to ensure all elements are *somewhere* in the log (all conflicting plugins are mentioned).

Also reworked the log message based on your suggestion.